examples/nbody/nbody.cpp

```diff
@@ -290,6 +293,8 @@ struct UpdateKernel
 #endif // NBODY_USE_SHARED
                         ts
                     );
+
+            syncBlockThreads(acc);
```

Comment:
I think we could avoid this barrier, in case we don't use the shared memory.

Reply:
Is this second barrier at the end needed at all? I understand the first barrier: It may be that some warp starts working on shared memory which is not filled by another, but after the shared memory is filled, everything should be save?

Reply:
Yes I think so, because some threads maybe already overwrite the shared memopry in the next round. Though we could move this barrier at the begin of the loop before line 272.

Reply:
Ah, I overlooked the outer loop because it ended with `};` instead of `}`. >.< You are right.

Reply:
One more reason to use `{}` for the loop body.

Reply:
What do you mean? In general a loop **should** end with `}`, not `};`, so my code has indeed a style fault, but if you have the comment of @ax3l in mind: I don't like one-line-loops to be encapsulated. :roll_eyes:

Reply:
yeah, for all reasonable cases. nobody could anticipate what you wrote *here* :D